src/main/java/org/spongepowered/common/mixin/core/server/MixinPlayerList.java

```diff
@@ -529,9 +529,31 @@ public EntityPlayerMP recreatePlayerEntity(EntityPlayerMP playerIn, int targetDi
         // Sponge - Vanilla does this before recreating the player entity. However, we need to determine the bed location
         // before respawning the player, so we know what dimension to spawn them into. This means that the bed location must be copied
         // over to the new player
+        // [Extended, Revised, 28.01.2018, MC 1.12.2, API 7.1.0] Fix for: https://github.com/SpongePowered/SpongeCommon/issues/1719
+        // First revision: Xakep_SDK, 28.01.18. MC Version: 1.12.2, API version: 7.1.0
+        // START REVISION
+        // This gets target world bed spawn location
+        // and saves original player dimension for RespawnPlayerEvent event.
+        // We're getting target world because of:
+        // 1. When we're calling EntityUtil#getPlayerRespawnLocation(EntityPlayerMP playerIn, @Nullable WorldServer targetWorld);
+        // at the end of the method we have if(bedPos != null) ...
+        // blockPos - actual bed location for target world, that we got here.
```

Comment:
This point doesn't make much sense. I'm not sure where the `if (bedPos != null)` comes from, unless you're referencing the next couple of lines.

Reply:
This refers to code inside revision block in EntityUtil